frontend/src/components/custom-tools/prompts-reorder/PromptsReorder.jsx

```diff
@@ -0,0 +1,171 @@
+  const { setAlertDetails } = useAlertStore();
+  const handleException = useExceptionHandler();
+
+  useEffect(() => {
+    if (!isOpen) {
+      setListOfPrompts([]);
+      return;
+    }
+
+    const prompts = details?.prompts || [];
+    setListOfPrompts(
+      prompts.map((prompt) => ({
+        prompt_key: prompt.prompt_key,
+        prompt_id: prompt.prompt_id,
+        sequence_number: prompt.sequence_number,
+      }))
+    );
+  }, [isOpen, details]);
+
+  const movePrompt = useCallback(
+    (fromIndex, toIndex) => {
+      if (fromIndex === toIndex) return;
+
+      // Store the previous state if not already stored
+      if (!previousListOfPrompts.current?.length) {
+        previousListOfPrompts.current = listOfPrompts;
+      }
+
+      setListOfPrompts((prevPrompts) => {
+        const updatedPrompts = [...prevPrompts];
+        const [movedItem] = updatedPrompts.splice(fromIndex, 1);
+        updatedPrompts.splice(toIndex, 0, movedItem);
+        return updatedPrompts;
+      });
+    },
+    [listOfPrompts]
+  );
+
+  const handleDropSuccess = useCallback((data) => {
+    const responseData = data || [];
+    const sequenceMap = responseData.reduce((acc, item) => {
+      acc[item.id] = item.sequence_number;
+      return acc;
+    }, {});
+
+    setListOfPrompts((prevPrompts) => {
+      const updatedPrompts = prevPrompts.map((prompt) => {
+        const newSequenceNumber = sequenceMap[prompt.prompt_id];
+        if (newSequenceNumber !== undefined) {
+          return { ...prompt, sequence_number: newSequenceNumber };
+        }
+        return prompt;
+      });
+
+      // Sort the prompts based on new sequence numbers
+      updatedPrompts.sort((a, b) => a.sequence_number - b.sequence_number);
+      return updatedPrompts;
+    });
+
+    previousListOfPrompts.current = [];
+  }, []);
+
+  const handleDropError = useCallback(
+    (err) => {
+      // Revert back the listOfPrompts state
+      setListOfPrompts(previousListOfPrompts.current);
+      previousListOfPrompts.current = [];
+      setAlertDetails(handleException(err, "Failed to reorder the prompts"));
+    },
+    [handleException, setAlertDetails]
+  );
+
+  const onDrop = useCallback(
+    async (fromIndex, toIndex) => {
+      if (fromIndex === toIndex) return;
+
+      updateReorderedStatus(true);
+
+      const body = {
+        start_sequence_number:
+          previousListOfPrompts.current[fromIndex]?.sequence_number,
+        end_sequence_number:
+          previousListOfPrompts.current[toIndex]?.sequence_number,
+        prompt_id: previousListOfPrompts.current[fromIndex]?.prompt_id,
+      };
+
+      const requestOptions = {
+        method: "POST",
+        url: `/api/v1/unstract/${sessionDetails?.orgId}/prompt-studio/prompt/reorder/`,
+        headers: {
+          "X-CSRFToken": sessionDetails?.csrfToken,
+          "Content-Type": "application/json",
+        },
+        data: body,
+      };
+
+      try {
```

Comment:
Is there any specific reason why we're using try catch instead of `.then` just to be uniform across the application.

Reply:
@jagadeeswaran-zipstack  I refactored the code to use a try-catch approach to resolve the sonar issue related to function nested more than four levels deep.